src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -26,11 +22,16 @@ impl ScalarFunc {
     ) -> Result<Option<Cow<'a, [u8]>>> {
         let t: Cow<'a, Time> = try_opt!(self.children[0].eval_time(ctx, row));
         if t.invalid_zero() {
-            return handle_incorrect_datetime_error(ctx, t).map(|_| None);
+            return ctx
+                .handle_invalid_time_error(Error::incorrect_datetime_value(&format!("{}", t)))
```

Comment:
`&t.to_string()` ?

But given the repetition I'd create a function like 

```rust
fn ensure_valid(ctx: ..., t: &Time) -> Result<()> {
    if t.invalid_zero() { return ... }
    Ok(())
}

...

ensure_valid(ctx, &t)?;
```

Reply:
We will refine errors later and the code shall be simplified then :)